pkg/signup/service/signup_service_test.go

```diff
@@ -1608,6 +1608,74 @@ func (s *TestSignupServiceSuite) TestGetSignupReadyConditionStatus() {
 	}
 }
 
+func (s *TestSignupServiceSuite) TestGetSignupBannedUserEmail() {
+	// given
+	s.ServiceConfiguration(configuration.Namespace(), true, "", 5)
+
+	us := s.newBannedUserSignup()
+	err := s.FakeUserSignupClient.Tracker.Add(us)
+	require.NoError(s.T(), err)
+
+	bannedUserID, err := uuid.NewV4()
+	require.NoError(s.T(), err)
+
+	err = s.FakeBannedUserClient.Tracker.Add(&toolchainv1alpha1.BannedUser{
+		TypeMeta: v1.TypeMeta{},
+		ObjectMeta: v1.ObjectMeta{
+			Name:      bannedUserID.String(),
+			Namespace: configuration.Namespace(),
+			Labels: map[string]string{
+				toolchainv1alpha1.BannedUserEmailHashLabelKey: "a7b1b413c1cbddbcd19a51222ef8e20a",
+			},
+		},
+		Spec: toolchainv1alpha1.BannedUserSpec{
+			Email: "jsmith@gmail.com",
+		},
+	})
+	require.NoError(s.T(), err)
+
+	rr := httptest.NewRecorder()
+	ctx, _ := gin.CreateTestContext(rr)
+	ctx.Set(context.UsernameKey, "jsmith")
+	ctx.Set(context.SubKey, us.Spec.IdentityClaims.UserID)
+	ctx.Set(context.EmailKey, "jsmith@gmail.com")
+
+	// when
+	response, err := s.Application.SignupService().GetSignup(ctx, us.Spec.IdentityClaims.UserID, us.Spec.IdentityClaims.PreferredUsername)
```

Comment:
It looks like if you comment out the code that adds the BannedUser to the `FakeBannedUserClient` then it doesn't cause the test to fail. I thought the test should fail on the `require.Nil(s.T(), response)` if there's no BannedUser.

Probably good to change the test to do 2 calls to GetSignup:

1. With just UserSignup and verify `require.NotNil(s.T(), response)`
2. With UserSignup and BannedUser and verify `require.Nil(s.T(), response)`

Reply:
Yeah, good catch, seems like the usersignup is not found , so it doesn't even get to the ban check anymore 🤔 

I think we may want to return a specific reason in the response instead of `nil` as we are discussing the the other thread here https://github.com/codeready-toolchain/registration-service/pull/404#discussion_r1491414417. This will make the test more reliable as well.